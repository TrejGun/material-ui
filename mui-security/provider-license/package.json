{
  "author": {
    "email": "trejgun+gemunion@gmail.com",
    "name": "TrejGun",
    "url": "https://gemunion.io/"
  },
  "bugs": {
    "url": "https://github.com/gemunion/mui-packages/issues"
  },
  "dependencies": {
    "@gemunion/license-messages": "0.0.3",
    "@gemunion/types-license": "0.0.3"
  },
  "description": "Gemunion license provider",
  "devDependencies": {
    "react": "17.0.2"
  },
  "engines": {
    "node": ">=12.13"
  },
  "keywords": [],
  "license": "UNLICENSED",
  "main": "./dist/index.js",
  "name": "@gemunion/provider-license",
  "peerDependencies": {
    "react": "^17"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/gemunion/mui-packages"
  },
  "scripts": {
    "build": "tsc",
    "lint": "eslint . --fix",
    "prebuild": "rimraf dist/",
    "prepare": "npm run build",
    "preversion": "npm run build && npm run test",
    "test": "exit 0"
  },
  "sideEffects": false,
  "title": "-=Gemun Ion Studio=-",
  "types": "./dist/index.d.ts",
  "version": "0.0.1"
}
