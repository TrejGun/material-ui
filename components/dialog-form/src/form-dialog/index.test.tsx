import { FC, ReactNode } from "react";
import { IntlProvider } from "react-intl";
import { cleanup, render } from "@testing-library/react";
import { createTheme, ThemeProvider } from "@mui/material";
import { MemoryRouter } from "react-router-dom";
import "whatwg-fetch";

import { LicenseProvider } from "@gemunion/provider-license";

import { FormDialog } from "./";

afterEach(cleanup);

const i18n = {
  "dialog.title": "Please fill the form",
  "form.buttons.cancel": "Cancel",
  "form.buttons.ok": "Ok",
  "form.hints.prompt": "Prompt",
};

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return (
    <LicenseProvider licenseKey={process.env.STORYBOOK_GEMUNION_LICENSE}>
      <MemoryRouter>
        <ThemeProvider theme={createTheme()}>
          <IntlProvider locale="en" messages={i18n}>
            {children}
          </IntlProvider>
        </ThemeProvider>
      </MemoryRouter>
    </LicenseProvider>
  );
};

describe("<FormDialog />", () => {
  it("renders closed dialog", () => {
    const container = document.createElement("div");
    document.body.append(container);

    const props = {
      container,
      open: false,
      onConfirm: jest.fn(),
      onCancel: jest.fn(),
      children: "some text",
      message: "dialog.title",
      initialValues: {
        id: 1,
        title: "Title",
      },
    };

    const { asFragment } = render(<FormDialog {...props} />, { container, wrapper: AllTheProviders });

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders open dialog", () => {
    const container = document.createElement("div");
    document.body.append(container);

    const props = {
      container,
      open: true,
      onConfirm: jest.fn(),
      onCancel: jest.fn(),
      children: "some text",
      message: "dialog.title",
      initialValues: {
        id: 1,
        title: "Title",
      },
    };

    const { asFragment } = render(<FormDialog {...props} />, { container, wrapper: AllTheProviders });

    expect(asFragment()).toMatchSnapshot();
  });
});
