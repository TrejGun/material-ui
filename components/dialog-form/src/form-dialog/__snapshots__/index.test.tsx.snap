// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormDialog /> renders closed dialog 1`] = `<DocumentFragment />`;
